chsdi/__init__.py

```diff
@@ -21,12 +25,30 @@ def cleanup(request):
     return session
 
 
+class WsgiSchemeAdaptedRequest(Request):
+    # TODO: remove this class, once apache is removed
+    # (https://jira.swisstopo.ch/browse/BGDIINF_SB-2486).
+    # Afterwards this can be solved similarly to our flask services using
+    # e.g. gunicorn.
+
+    def __init__(self, environ, **kwargs):
+        logger.warning("********************** BEFORE: %s", environ)
+        if "HTTP_CLOUDFRONT_FORWARDED_PROTO" in environ:
+            environ["wsgi.url_scheme"] = environ["HTTP_CLOUDFRONT_FORWARDED_PROTO"]
+        elif "HTTP_X_FORWARDED_PROTO" in environ:
+            environ["wsgi.url_scheme"] = environ["HTTP_X_FORWARDED_PROTO"]
+        super().__init__(environ, **kwargs)
+        logger.warning("*********************** AFTER: %s", environ)
```

Comment:
Oops do not Forget to clean up the code here and to remove the logger to speed up the code.

Reply:
Oops, thanks. They came with the last commit, I guess. Will remove them of course :wink: